### PR TITLE
dataplex metadata integration

### PR DESCRIPTION
**Pull request by AmandeepSinghCS**

### CODE REVIEW

File: tests/integration/test_dataplex_metadata.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import pytest
+from clouddq.integration.dataplex.clouddq_dataplex import CloudDqDataplexClient
+import logging
+
+logger = logging.getLogger(__name__)
+
+class TestMetadataIntegration:
+
+    @pytest.fixture
+    def test_dq_dataplex_client(self,
```
Comment (thinhha) [inline tests/integration/test_dataplex_metadata.py:24]: could you move this client fixture to conftest and reuse it across test_dataplex_metadata.py and test_dataplex_task.py?

File: tests/integration/test_dataplex_metadata.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    )
+    def test_dataplex_metadata_get_entity(self,
+                                            test_dq_dataplex_client,
+                                            gcp_dataplex_zone_id,
+                                            entity_id,
+                                            params,):
+
+        response = test_dq_dataplex_client.get_entity(zone_id=gcp_dataplex_zone_id,
+                                                      entity_id=entity_id,
+                                                      params=params)
```
Comment (thinhha) [inline tests/integration/test_dataplex_metadata.py:71]: remove params here

File: clouddq/BUILD [hunk 3]
```diff
@@ -77,6 +77,7 @@ py_library(
     visibility = ["//tests:__subpackages__"],
     deps = [
         requirement("jinja2"),
+        requirement("pyyaml"),
```
Comment (thinhha) [inline clouddq/BUILD:80]: why is this needed?

File: clouddq/classes/dataplex_entity.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    asset: str
+    dataPath: str
+    system: str
+    format: dict
+    schema: DataplexEntitySchema
+
+
+    @property
+    def project_id(self):
+        return self.name.split("/")[1:2][0]
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity.py:39]: why not just self.name.split("/")[1]?

File: clouddq/classes/dataplex_entity.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        """
+
+        Args:
+          self: DataplexEntity:
+
+        Returns:
+
+        """
+
+        return dict(self.to_dict().get(self.id))
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity.py:144]: is this working? we should delete dict_values() if we're not using it.

File: clouddq/classes/dataplex_entity_schema.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    @classmethod
+    def from_dict(cls: DataplexEntitySchema, kwargs: dict) -> DataplexEntitySchema:
+
+        fields = kwargs.get("fields")
+        assert_not_none_or_empty(value=fields,
+                 error_msg=f"Fields: must define non-empty value: 'fields'.")
+
+        return DataplexEntitySchema(
+            fields=fields,
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema.py:37]: does this fail if fields is not a list of DataplexEntitySchemaField?

File: clouddq/classes/dataplex_entity_schema_field.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+        Returns:
+
+        """
+
+        return asdict(self)
+
+
+    @classmethod
+    def from_dict(cls: DataplexEntitySchemaField, kwargs: dict) -> DataplexEntitySchemaField:
```
Comment (thinhha) [inline clouddq/classes/dataplex_entity_schema_field.py:42]: where is this called?

File: clouddq/classes/dataplex_entity_schema.py (hunk 6 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema.py:37]: In theory this should not fail as the list can have one or multiple DataplexEntitySchemaField objects

File: clouddq/classes/dataplex_entity.py (hunk 5 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity.py:144]: deleted as we are not using it.

File: clouddq/classes/dataplex_entity_schema_field.py (hunk 7 above)
Comment (AmandeepSinghCS) [reply clouddq/classes/dataplex_entity_schema_field.py:42]: I created this for testing purpose, if we want to mock schema for the dataplex entity.
Not getting called anywhere.

File: scripts/bazel_test.sh [hunk 8]
```diff
@@ -67,6 +70,9 @@ function bazel_test() {
       --test_env DATAPLEX_TARGET_BQ_DATASET="${DATAPLEX_TARGET_BQ_DATASET:-}" \
       --test_env DATAPLEX_TARGET_BQ_TABLE="${DATAPLEX_TARGET_BQ_TABLE:-}" \
       --test_env DATAPLEX_TASK_SA="${DATAPLEX_TASK_SA:-}" \
+      --test_env DATAPLEX_ZONE_ID="${DATAPLEX_ZONE_ID}" \
+      --test_env DATAPLEX_BUCKET_NAME="${DATAPLEX_BUCKET_NAME}" \
+      --test_env DATAPLEX_BIGQUERY_DATASET_ID="${DATAPLEX_BIGQUERY_DATASET_ID}" \
```
Comment (thinhha) [inline scripts/bazel_test.sh:75]: can you make these optional? e.g. ${DATAPLEX_ZONE_ID:-}
